src/unitxt/task.py

```diff
@@ -99,9 +106,46 @@ def check_metrics_type(self) -> None:
                 f"metric's prediction type ({metric_prediction_type}) are different."
             )
 
+    def verify_defaults(self):
+        if self.defaults:
+            if not isinstance(self.defaults, dict):
+                raise ValueError(
+                    f"If specified, the 'defaults' must be a dictionary, "
+                    f"however, '{self.defaults}' was provided instead, "
+                    f"which is of type '{type(self.defaults)}'."
+                )
+
+            for default_name, default_value in self.defaults.items():
+                assert isinstance(default_name, str), (
+                    f"If specified, all keys of the 'defaults' must be strings, "
+                    f"however, the key '{default_name}' is of type '{type(default_name)}'."
+                )
+
+                val_type = self.inputs.get(default_name) or self.outputs.get(
```

Comment:
This assumes the task fields  are in the new format (dictionary) and not in the old list format (still used by some people).  Perhaps add a check , that inputs and outputs are dict and fail with "default params are only supported when input and output fields are defined as { "field_name1" : "field_type1" , ....} and not [ "field_name1" ... ] .

Reply:
But this happens at the end of the `verify` method which converts both `inputs` and `outputs` to directories anyway

Reply:
I see.  IF so, please ignore this.